saleor/graphql/views.py

```diff
@@ -176,8 +176,13 @@ def handle_query(self, request: HttpRequest) -> JsonResponse:
             span.set_tag("http.useragent", request.META.get("HTTP_USER_AGENT", ""))
             span.set_tag("span.type", "web")
 
-            request_ips = request.META.get(settings.REAL_IP_ENVIRON, "")
-            for ip in request_ips.split(","):
+            request_ips = []
+            for header in settings.REAL_IP_ENVIRON:
+                ips = request.META.get(header)
+                if ips:
+                    request_ips.extend(ips.split(","))
+
+            for ip in request_ips:
                 if is_valid_ipv4(ip):
                     span.set_tag(opentracing.tags.PEER_HOST_IPV4, ip)
```

Comment:
You are taking the first valid IP from the list as the span tag instead of taking the first valid of all headers. I would suggest to use the `REAL_IP_ENVIRON[0]` as `PEER_HOST_IPV*` and the rest as `ip_<header>`

Reply:
I left the value of `ip_<header>` tag as raw header values to store traces of suspicious values